### PR TITLE
Digital Summative Assessment Button

### PR DESCRIPTION
**Pull request by msquance-stem**
## Status

* Current Status: Ready for tech review
* Review App link(s): *populate this with links to the relevant parts of the review app*
* Closes: https://github.com/NCCE/teachcomputing.org-issues/issues/2530
* 
## Review progress:

- [ ] Browser tested
- [ ] Front-end review completed
- [ ] Tech review completed

## What's changed?

Added the new card for external links to the unit page, this has been done as a mixed component that supports files and external links so ordering can be changed easily. Created a component to support the external link in the style of a file card so they look similar. Also made the changes to the GRAPHQL query to pull the new digital summative assessment from curriculum.

### CODE REVIEW

File: app/components/mixed_cards_component.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class MixedCardsComponent < ViewComponent::Base
```
Comment (Caleb-T-Owens) [inline app/components/mixed_cards_component.rb:3]: Would it be better to just have one array which is `cards` which contains objects like:
```rb
{
  type: :file,
  details: { ... }
}
```

So you can then iterate over them all like:
```
<% @cards.each do |card| %>
  <% case card_group[:type] %>
    <% when :file %>
      <%= render FileCardComponent.new(card[:details], title: @title, tracking: @tracking) %>
    <% when :external_link %>
      <%= render ExternalLinkCardComponent.new(card[:details]) %>
    <% end %>
  <% end %>
<% end %>
```

Meaning that we're not restricted by any ordering so if we so desired the external_link could be in the middle of the files

**Review by Caleb-T-Owens — COMMENTED** [review]
Other than a minor comment, looking pretty good

File: app/components/mixed_cards_component.rb (hunk 1 above)
Comment (msquance-stem) [reply app/components/mixed_cards_component.rb:3]: Yeah its probably clearer that way. Will make the change

**Review by Caleb-T-Owens — APPROVED** [review]
LGTM 🚀

**Pull request merged**